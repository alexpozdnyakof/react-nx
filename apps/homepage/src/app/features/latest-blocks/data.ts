import { BlockShort } from '@blockchain/api-interfaces';

export const BLX_DATA: Array<BlockShort> = [
  ...new Array(5).fill({
    index: 75260722,
    time: 1663680799353,
    size: 1200,
    txsCount: 723,
    occupancy: 71,
  }),
];
