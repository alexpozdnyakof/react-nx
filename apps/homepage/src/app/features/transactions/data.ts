export const DEMO_DATA = [
  {
    hash: '123-qwe',
    time: 1663353730855,
    amountUsd: 1000,
    amountBtc: 1000,
  },
  {
    hash: '123-qwe',
    time: 1663353730855,
    amountUsd: 1000,
    amountBtc: 1000,
  },
  {
    hash: '123-qwe',
    time: 1663353730855,
    amountUsd: 1000,
    amountBtc: 1000,
  },
];
