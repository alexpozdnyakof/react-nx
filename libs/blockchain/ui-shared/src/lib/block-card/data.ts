const BLX_DATA = [
  {
    index: '752,722',
    time: new Intl.DateTimeFormat('en-US', {
      year: 'numeric',
      month: 'numeric',
      day: 'numeric',
      hour: 'numeric',
      minute: 'numeric',
      second: 'numeric',
      hour12: false,
    }).format(Date.now()),
    size: '1.2 Mb',
    txsCount: 723,
    occupancy: 71,
  },
];

export { BLX_DATA };
