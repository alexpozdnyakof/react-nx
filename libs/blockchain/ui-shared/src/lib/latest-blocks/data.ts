const BLXS_DATA = [
  ...new Array(5).fill({
    index: '752,722',
    time: new Intl.DateTimeFormat('en-US', {
      year: 'numeric',
      month: 'numeric',
      day: 'numeric',
      hour: 'numeric',
      minute: 'numeric',
      second: 'numeric',
      hour12: false,
    }).format(Date.now()),
    size: '1.2 Mb',
    txsCount: 723,
    occupancy: '71%',
  }),
];
export { BLXS_DATA };
