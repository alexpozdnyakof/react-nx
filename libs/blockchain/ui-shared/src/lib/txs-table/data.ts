const TXS_DATA = [
  ...new Array(10).fill(['ccf3e-2cd36', Date.now(), 0.76729409, 15321.09]),
].map(([hash, time, amountBtc, amountUsd]) => ({
  hash,
  time,
  amountBtc,
  amountUsd,
}));

export { TXS_DATA };
