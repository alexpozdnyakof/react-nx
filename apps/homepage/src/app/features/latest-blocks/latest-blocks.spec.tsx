import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import testRenderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { BLX_DATA } from './data';
import LatestBlocks from './latest-blocks';
import reducer, { load, loadSuccess } from './slice';
import * as storeHooks from '../../store-hooks'

describe('latest blocks', () => {
  const store = configureStore({reducer: {latestBlocks: reducer}});
  const WithStore = () => (<Provider store={store}><LatestBlocks/></Provider>);

  it('should render succesfully', () => {
    const base = render(<WithStore/>)
    expect(base).toBeDefined()
  })

  it('should render correctly with transactions', ()=>{
    store.dispatch(loadSuccess(BLX_DATA))
    const tree = testRenderer.create(<WithStore/>).toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('should dispatch load', ()=>{
    const dispatchMock = jest.fn()
    jest.spyOn(storeHooks, 'useAppDispatch').mockReturnValue(dispatchMock);

    render(<WithStore/>);
    expect(dispatchMock).toHaveBeenCalledWith(load());
  })
})
