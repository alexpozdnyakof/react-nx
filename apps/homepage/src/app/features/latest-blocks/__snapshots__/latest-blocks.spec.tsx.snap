// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`latest blocks should render correctly with transactions 1`] = `
<div
  className="Widget"
>
  <div
    className="Widget-Header"
  >
    <img
      alt="widget symbol"
      className="Widget-Symbol"
      src="https://www.blockchain.com/explorer/_next/static/media/btc.a6006067.png"
    />
    <div
      className="Widget-Info"
    >
      <a
        className="Link"
        data-testid="title-link"
        href="https://www.blockchain.com/btc/unconfirmed-transactions"
      >
        <h1
          className="Widget-Title"
        >
          Latest Blocks
        </h1>
      </a>
      <h2
        className="Widget-SubTitle"
      >
        Bitcoin
      </h2>
    </div>
    <div
      className="Spacer"
    />
    <a
      className="Link"
      data-testid="arrow-link"
      href="https://www.blockchain.com/btc/unconfirmed-transactions"
    >
      <span
        className="Widget-Arrow"
      >
        →
      </span>
    </a>
  </div>
  <div
    className="Block-Container"
  >
    <div
      className="BlockCard"
      data-testid="block-card"
    >
      <div
        className="Block-Indicator"
      >
        <div
          className="Indicator"
        >
          <div
            className="Payload"
            data-testid="occupancy-payload"
            style={
              Object {
                "height": "71%",
              }
            }
          >
            <div
              className="Payload-Fill"
            />
          </div>
        </div>
      </div>
      <div
        className="Block-Description"
      >
        <div
          className="Description-Item"
        >
          <span
            className="Block-Title"
          >
            75260722
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Time"
          >
            9/20/2022, 20:33:19
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Size"
          >
            723tx
              • 
            NaNmb
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Block-Container"
  >
    <div
      className="BlockCard"
      data-testid="block-card"
    >
      <div
        className="Block-Indicator"
      >
        <div
          className="Indicator"
        >
          <div
            className="Payload"
            data-testid="occupancy-payload"
            style={
              Object {
                "height": "71%",
              }
            }
          >
            <div
              className="Payload-Fill"
            />
          </div>
        </div>
      </div>
      <div
        className="Block-Description"
      >
        <div
          className="Description-Item"
        >
          <span
            className="Block-Title"
          >
            75260722
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Time"
          >
            9/20/2022, 20:33:19
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Size"
          >
            723tx
              • 
            NaNmb
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Block-Container"
  >
    <div
      className="BlockCard"
      data-testid="block-card"
    >
      <div
        className="Block-Indicator"
      >
        <div
          className="Indicator"
        >
          <div
            className="Payload"
            data-testid="occupancy-payload"
            style={
              Object {
                "height": "71%",
              }
            }
          >
            <div
              className="Payload-Fill"
            />
          </div>
        </div>
      </div>
      <div
        className="Block-Description"
      >
        <div
          className="Description-Item"
        >
          <span
            className="Block-Title"
          >
            75260722
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Time"
          >
            9/20/2022, 20:33:19
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Size"
          >
            723tx
              • 
            NaNmb
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Block-Container"
  >
    <div
      className="BlockCard"
      data-testid="block-card"
    >
      <div
        className="Block-Indicator"
      >
        <div
          className="Indicator"
        >
          <div
            className="Payload"
            data-testid="occupancy-payload"
            style={
              Object {
                "height": "71%",
              }
            }
          >
            <div
              className="Payload-Fill"
            />
          </div>
        </div>
      </div>
      <div
        className="Block-Description"
      >
        <div
          className="Description-Item"
        >
          <span
            className="Block-Title"
          >
            75260722
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Time"
          >
            9/20/2022, 20:33:19
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Size"
          >
            723tx
              • 
            NaNmb
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Block-Container"
  >
    <div
      className="BlockCard"
      data-testid="block-card"
    >
      <div
        className="Block-Indicator"
      >
        <div
          className="Indicator"
        >
          <div
            className="Payload"
            data-testid="occupancy-payload"
            style={
              Object {
                "height": "71%",
              }
            }
          >
            <div
              className="Payload-Fill"
            />
          </div>
        </div>
      </div>
      <div
        className="Block-Description"
      >
        <div
          className="Description-Item"
        >
          <span
            className="Block-Title"
          >
            75260722
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Time"
          >
            9/20/2022, 20:33:19
          </span>
        </div>
        <div
          className="Description-Item"
        >
          <span
            className="Block-Size"
          >
            723tx
              • 
            NaNmb
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
